Jenkinsfile

```diff
@@ -47,48 +70,89 @@ builders['msvc2015_64'] = { node('windows'){
+                    bat script: "type error_and_warnings.txt"
+                    bat script: "cmake --build . --target lint --config ${buildConfiguration} -- /maxcpucount:8 > lint_error_and_warnings.txt"
+                    bat script: "type lint_error_and_warnings.txt"
+                    bat script: "cpack -D CPACK_PACKAGE_FILE_NAME=veles-${version} -G \"${cpack_generator}\" -C ${buildConfiguration}"
+                  }
+                  junit allowEmptyResults: true, keepLongStdio: true, testResults: '**/results.xml'
+                  step([$class: 'ArtifactArchiver', artifacts: "build_${version}/veles-${version}.*", fingerprint: true])
+                  step([$class: 'WarningsPublisher',
+                    canComputeNew: false,
+                    canResolveRelativePaths: false,
+                    defaultEncoding: '',
+                    excludePattern: '',
+                    healthy: '',
+                    includePattern: '',
+                    messagesPattern: '',
+                    parserConfigurations: [
+                      [parserName: 'MSBuild', pattern: "build_${version}/error_and_warnings.txt"],
+                      [parserName: 'clang-tidy-msvc', pattern: "build_${version}/lint_error_and_warnings.txt"]
+                    ],
+                    unHealthy: ''
+                  ])
+
+                  step([$class: 'WarningsPublisher', canComputeNew: false, canResolveRelativePaths: false, defaultEncoding: '',
+                        excludePattern: '', healthy: '', includePattern: '', messagesPattern: '',
+                        parserConfigurations: [[parserName: 'MSBuild', pattern: "build_${version}/error_and_warnings.txt"]], unHealthy: ''])
+                  bat(script: "rd /s /q build_${version}", returnStatus: true)
+                }
+              }  catch (error) {
+                post_stage_failure(env.JOB_NAME, "windows-msvc2015-x64", env.BUILD_ID, env.BUILD_URL)
+                bat script: "type build_${version}\\*error_and_warnings.txt"
+                throw error
               }
-              bat(script: "rd /s /q build_${version}", returnStatus: true)
-              bat script: "md build_${version}", returnStatus: true
-              dir ("build_${version}") {
-                bat script: "cmake -DCMAKE_PREFIX_PATH=%CMAKE_PREFIX_PATH% -DGOOGLETEST_SRC_PATH=%GOOGLETEST_DIR% -DEMBED_PYTHON_ARCHIVE_PATH=%EMBED_PYTHON_ARCHIVE_PATH% -DVCREDIST_BINARY=\"${vcredist_binary}\" -DOPENSSL_DLL_DIR=${tool 'openssl-64'} -G \"${generator}\" ..\\"
-                bat script: "cmake --build . --config ${buildConfiguration} > error_and_warnings.txt"
-                bat script: "type error_and_warnings.txt"
-                bat script: "cpack -D CPACK_PACKAGE_FILE_NAME=veles-${version} -G \"${cpack_generator}\" -C ${buildConfiguration}"
+            }
+          },
+          "python tests - Windows 10": {
```

Comment:
It is where we run our tests. Actually, it is not Windows 10, but Windows 2016, so I just removed version reference.